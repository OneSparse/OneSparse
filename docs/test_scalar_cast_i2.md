# Scalar `smallint` Casting

This documentation is also tests for the code, the examples below
show the literal output of these statements from Postgres.

Some setup to make sure warnings are shown, and that the extension
is installed.
``` postgres-console
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
```
These casting functions cast the Postgres type `smallint` to the
GraphBLAS scalar type `GrB_INT16`.
``` postgres-console
select cast(-1::smallint as scalar);
 scalar 
--------
 i2:-1
(1 row)

select cast(0::smallint as scalar);
 scalar 
--------
 i2:0
(1 row)

select cast(1::smallint as scalar);
 scalar 
--------
 i2:1
(1 row)

```
These tests cast back from the scalar type `GrB_INT16` to the
Postgres type `smallint`
``` postgres-console
select cast(-1::smallint::scalar as smallint);
ERROR:  operator does not exist: - scalar
LINE 1: select cast(-1::smallint::scalar as smallint);
                    ^
HINT:  No operator matches the given name and argument type. You might need to add an explicit type cast.
select cast(0::smallint::scalar as  smallint);
 int2 
------
    0
(1 row)

select cast(1::smallint::scalar as scalar);
 scalar 
--------
 i2:1
(1 row)

```