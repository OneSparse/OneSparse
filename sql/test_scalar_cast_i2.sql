-- # Scalar `smallint` Casting
--
-- This documentation is also tests for the code, the examples below
-- show the literal output of these statements from Postgres.
--
-- Some setup to make sure warnings are shown, and that the extension
-- is installed.
set client_min_messages = 'WARNING';
create extension if not exists onesparse;

-- These casting functions cast the Postgres type `smallint` to the
-- GraphBLAS scalar type `GrB_INT16`.

select cast(-1::smallint as scalar);
select cast(0::smallint as scalar);
select cast(1::smallint as scalar);

-- These tests cast back from the scalar type `GrB_INT16` to the
-- Postgres type `smallint`

select cast(-1::smallint::scalar as smallint);
select cast(0::smallint::scalar as  smallint);
select cast(1::smallint::scalar as scalar);
