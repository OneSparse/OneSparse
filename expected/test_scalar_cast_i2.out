set client_min_messages = 'WARNING';
create extension if not exists onesparse;
set search_path = 'onesparse';
select cast(-1::smallint as scalar);
 scalar 
--------
 i2:-1
(1 row)

select cast(0::smallint as scalar);
 scalar 
--------
 i2:0
(1 row)

select cast(1::smallint as scalar);
 scalar 
--------
 i2:1
(1 row)

select cast(-1::smallint::scalar as smallint);
ERROR:  operator does not exist: - scalar
LINE 1: select cast(-1::smallint::scalar as smallint);
                    ^
HINT:  No operator matches the given name and argument type. You might need to add an explicit type cast.
select cast(0::smallint::scalar as  smallint);
 int2 
------
    0
(1 row)

select cast(1::smallint::scalar as scalar);
 scalar 
--------
 i2:1
(1 row)

